executor/show.go

```diff
@@ -590,13 +590,29 @@ func (e *ShowExec) fetchShowCreateTable() error {
 		return nil
 	}
 
+	tblCharset := tb.Meta().Charset
+	if len(tblCharset) == 0 {
+		tblCharset = mysql.DefaultCharset
+	}
+	tblCollate := tb.Meta().Collate
+	// Set default collate if collate is not specified.
+	if len(tblCollate) == 0 {
+		tblCollate = getDefaultCollate(tblCharset)
+	}
+
 	fmt.Fprintf(&buf, "CREATE TABLE %s (\n", escape(tb.Meta().Name, sqlMode))
 	var pkCol *table.Column
 	var hasAutoIncID bool
 	for i, col := range tb.Cols() {
 		fmt.Fprintf(&buf, "  %s %s", escape(col.Name, sqlMode), col.GetTypeDesc())
 		if col.Charset != "binary" {
-			fmt.Fprintf(&buf, " CHARSET %s COLLATE %s", col.Charset, col.Collate)
+			if col.Charset != tblCharset || col.HasExplictedCollation() {
+				fmt.Fprintf(&buf, " CHARSET %s", col.Charset)
```

Comment:
```suggestion
				fmt.Fprintf(&buf, " CHARSET %s COLLATE %s",  col.Charset, col.Collate)
```

Reply:
I think now what I write is more like what MYSQL writes, how do you think it ?